cli/cmd/check.go

```diff
@@ -55,22 +58,26 @@ non-zero exit code.`,
   linkerd check --proxy --namespace app`,
 		Args: cobra.NoArgs,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return configureAndRunChecks(stdout, options)
+			return configureAndRunChecks(cmd, stdout, options)
 		},
 	}
 
 	cmd.Args = cobra.NoArgs
 	cmd.PersistentFlags().StringVar(&options.versionOverride, "expected-version", options.versionOverride, "Overrides the version used when checking if Linkerd is running the latest version (mostly for testing)")
 	cmd.PersistentFlags().BoolVar(&options.preInstallOnly, "pre", options.preInstallOnly, "Only run pre-installation checks, to determine if the control plane can be installed")
-	cmd.PersistentFlags().BoolVar(&options.dataPlaneOnly, "proxy", options.dataPlaneOnly, "Only run data-plane checks, to determine if the data plane is healthy")
+	cmd.PersistentFlags().StringVar(&options.targetProxyResource, "proxy", options.targetProxyResource, "Only run data-plane checks, to determine if the data plane is healthy")
```

Comment:
I'd like for the target of the data plane check to be optional. In order to do that, we need to leave this as a boolean flag, and instead update the command itself to take an optional additional argument. Here's a diff of what that would look like against this file on master:

```diff
diff --git a/cli/cmd/check.go b/cli/cmd/check.go
index fcf10c21..09bb3ff4 100644
--- a/cli/cmd/check.go
+++ b/cli/cmd/check.go
@@ -53,13 +53,16 @@ non-zero exit code.`,
 
   # Check that the Linkerd data plane proxies in the "app" namespace are up and running
   linkerd check --proxy --namespace app`,
-		Args: cobra.NoArgs,
+		Args: cobra.MaximumNArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return configureAndRunChecks(stdout, options)
+			var target string
+			if len(args) == 1 {
+				target = args[0]
+			}
+			return configureAndRunChecks(stdout, options, target)
 		},
 	}
 
-	cmd.Args = cobra.NoArgs
 	cmd.PersistentFlags().StringVar(&options.versionOverride, "expected-version", options.versionOverride, "Overrides the version used when checking if Linkerd is running the latest version (mostly for testing)")
 	cmd.PersistentFlags().BoolVar(&options.preInstallOnly, "pre", options.preInstallOnly, "Only run pre-installation checks, to determine if the control plane can be installed")
 	cmd.PersistentFlags().BoolVar(&options.dataPlaneOnly, "proxy", options.dataPlaneOnly, "Only run data-plane checks, to determine if the data plane is healthy")
@@ -70,7 +73,10 @@ non-zero exit code.`,
 	return cmd
 }
 
-func configureAndRunChecks(w io.Writer, options *checkOptions) error {
+func configureAndRunChecks(w io.Writer, options *checkOptions, target string) error {
+	if target != "" {
+		options.dataPlaneOnly = true
+	}
 	err := options.validate()
 	if err != nil {
 		return fmt.Errorf("Validation error when executing check command: %v", err)
@@ -110,6 +116,7 @@ func configureAndRunChecks(w io.Writer, options *checkOptions) error {
 		KubeContext:           kubeContext,
 		APIAddr:               apiAddr,
 		VersionOverride:       options.versionOverride,
+		TargetProxyResource:   target,
 		RetryDeadline:         time.Now().Add(options.wait),
 	})
 
```

Reply:
hmm so you think my approach with `cmd.Flags().Changed("proxy")` is not correct? Hmm, yes, you're right, I just tested it. I don't know why am I testing this now and not earlier, sorry 🙈 For my solution it says "Error: flag needs an argument: --proxy"

I'll try your approach

Reply:
OK, I have some challenges making this work, I'll have to get back to this later.

Btw. will this work also in this way? `linkerd check --proxy deploy/web -n emojivoto` because I am afraid it won't

Reply:
@alenkacz I think that should work. Cobra allows args to be interspersed between flags like that.

Reply:
hmm when I did that change locally, cobra is telling me 'deployment/web' is unknown command. Any ideas until I dive deeper? As a part of that endeavor I wanted to update cobra anyway, it seems linkerd is using pretty old version. Thoughts on both?